src/vs/workbench/api/common/extensionHostMain.ts

```diff
@@ -128,7 +134,18 @@ export abstract class ErrorHandler {
 				extension = stackData?.extensionIdentifier;
 			}
 
-			if (extension) {
+			if (!extension) {
+				return;
+			}
+
+			if (errors.PendingMigrationError.is(err)) {
+				// report pending migration via the API deprecation service which (1) informs the extensions author during
+				// dev-time and (2) collects telemetry so that we can reach out too
+				const extensionDesc = extensionsRegistry.getExtensionDescription(extension);
+				if (extensionDesc) {
+					apiDeprecationService.report(err.name, extensionDesc, err.message);
```

Comment:
`apiDeprecationService.report` does not print any stack trace, if `extension.isUnderDevelopment` should we print the stack so that it assists with tracing back the deprecation ?

```suggestion
					apiDeprecationService.report(err.name, extensionDesc, err.message);
					if (extensionDesc.isUnderDevelopment) {
					   logService.error(err);
					 }
```